src/scheduler/node_info.c

```diff
@@ -4745,7 +4745,7 @@ reorder_nodes(node_info **nodes, resource_resv *resresv)
 	nptr = node_array;
 
 
-	if (last_node_name[0] == '\0')
+	if (last_node_name[0] == '\0' && nodes[0] != NULL)
 		strcpy(last_node_name, nodes[0]->name);
```

Comment:
Can you turn this into snprintf(last_node_name, sizeof(last_node_name), "%s", nodes[0]->name)?